engine/hatchery/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,307 @@
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(configK8s)
+	if err != nil {
+		return sdk.WrapError(err, "Cannot create new config")
+	}
+
+	h.k8sClient = clientset
+
+	if h.Config.KubernetesNamespace != apiv1.NamespaceDefault {
+		if _, err := clientset.CoreV1().Namespaces().Get(h.Config.KubernetesNamespace, metav1.GetOptions{}); err != nil {
+			ns := apiv1.Namespace{}
+			ns.SetName(h.Config.KubernetesNamespace)
+			if _, errC := clientset.CoreV1().Namespaces().Create(&ns); errC != nil {
+				return sdk.WrapError(errC, "Cannot create namespace %s in kubernetes", h.Config.KubernetesNamespace)
+			}
+		}
+	}
+
+	return nil
+}
+
+// getStartingConfig implements ConfigAccess
+func (h *HatcheryKubernetes) getStartingConfig() (*clientcmdapi.Config, error) {
+	defaultClientConfigRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	overrideCfg := clientcmd.ConfigOverrides{
+		AuthInfo: clientcmdapi.AuthInfo{
+			Username: h.Config.KubernetesUsername,
+			Password: h.Config.KubernetesPassword,
+			Token:    h.Config.KubernetesToken,
+		},
+	}
+
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(defaultClientConfigRules, &overrideCfg)
+	rawConfig, err := clientConfig.RawConfig()
+	if os.IsNotExist(err) {
+		return clientcmdapi.NewConfig(), nil
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return &rawConfig, nil
+}
+
+// CheckConfiguration checks the validity of the configuration object
+func (h *HatcheryKubernetes) CheckConfiguration(cfg interface{}) error {
+	hconfig, ok := cfg.(HatcheryConfiguration)
+	if !ok {
+		return fmt.Errorf("Invalid configuration")
+	}
+
+	if hconfig.API.HTTP.URL == "" {
+		return fmt.Errorf("API HTTP(s) URL is mandatory")
+	}
+
+	if hconfig.API.Token == "" {
+		return fmt.Errorf("API Token URL is mandatory")
+	}
+
+	if hconfig.Name == "" {
+		return fmt.Errorf("please enter a name in your local hatchery configuration")
+	}
+
+	if hconfig.KubernetesNamespace == "" {
+		return fmt.Errorf("please enter a valid kubernetes namespace")
+	}
+
+	if hconfig.KubernetesMasterURL == "" {
+		return fmt.Errorf("please enter a valid kubernetes master URL")
+	}
+
+	return nil
+}
+
+// Serve start the HatcheryKubernetes server
+func (h *HatcheryKubernetes) Serve(ctx context.Context) error {
+	hatchery.Create(h)
+	return nil
+}
+
+// ID must returns hatchery id
+func (h *HatcheryKubernetes) ID() int64 {
+	if h.hatch == nil {
+		return 0
+	}
+	return h.hatch.ID
+}
+
+//Hatchery returns hatchery instance
+func (h *HatcheryKubernetes) Hatchery() *sdk.Hatchery {
+	return h.hatch
+}
+
+//Client returns cdsclient instance
+func (h *HatcheryKubernetes) Client() cdsclient.Interface {
+	return h.client
+}
+
+//Configuration returns Hatchery CommonConfiguration
+func (h *HatcheryKubernetes) Configuration() hatchery.CommonConfiguration {
+	return h.Config.CommonConfiguration
+}
+
+// ModelType returns type of hatchery
+func (*HatcheryKubernetes) ModelType() string {
```

Comment:
this should be the type of worker model ?

Reply:
done